service/src/main/java/bio/terra/workspace/service/resource/controlled/cloud/azure/storage/DeleteAzureStorageStep.java

```diff
@@ -0,0 +1,65 @@
+package bio.terra.workspace.service.resource.controlled.cloud.azure.storage;
+
+import bio.terra.stairway.FlightContext;
+import bio.terra.stairway.Step;
+import bio.terra.stairway.StepResult;
+import bio.terra.stairway.StepStatus;
+import bio.terra.workspace.app.configuration.external.AzureConfiguration;
+import bio.terra.workspace.service.crl.CrlService;
+import bio.terra.workspace.service.workspace.flight.WorkspaceFlightMapKeys.ControlledResourceKeys;
+import bio.terra.workspace.service.workspace.model.AzureCloudContext;
+import com.azure.resourcemanager.storage.StorageManager;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * A step for deleting a controlled Azure Storage Account resource.
+ */
+public class DeleteAzureStorageStep implements Step {
+  private static final Logger logger = LoggerFactory.getLogger(CreateAzureStorageStep.class);
+  private final AzureConfiguration azureConfig;
+  private final CrlService crlService;
+  private final ControlledAzureStorageResource resource;
+
+  public DeleteAzureStorageStep(AzureConfiguration azureConfig, CrlService crlService,
+                                ControlledAzureStorageResource resource) {
+    this.crlService = crlService;
+    this.azureConfig = azureConfig;
+    this.resource = resource;
+  }
+
+  @Override
+  public StepResult doStep(FlightContext context) throws InterruptedException {
+    final AzureCloudContext azureCloudContext =
+        context
+            .getWorkingMap()
+            .get(ControlledResourceKeys.AZURE_CLOUD_CONTEXT, AzureCloudContext.class);
+
+    StorageManager manager = crlService.getStorageManager(azureCloudContext, azureConfig);
+    var azureResourceId =
+        String.format(
+            "/subscriptions/%s/resourceGroups/%s/providers/Microsoft.Storage/storageAccounts/%s",
+            azureCloudContext.getAzureSubscriptionId(),
+            azureCloudContext.getAzureResourceGroupId(),
+            resource.getStorageAccountName());
+    try {
+      logger.info("Attempting to delete Azure Storage Account " + azureResourceId);
+      manager.storageAccounts().deleteById(azureResourceId);
+      return StepResult.getStepResultSuccess();
+    } catch (Exception ex) {
```

Comment:
🤔 `Exception` seems overly broad here -- I wonder if we can limit this to just the [Azure exception hierarchy](https://docs.microsoft.com/en-us/java/api/overview/azure/core-readme?view=azure-java-stable#exception-hierarchy-with-azureexception)

Reply:
@ddietterich what do you think about this one? This is another place where I was just following the example of other deletion steps.

Reply:
I'm going to leave this as-is unless I hear back that I should make the exception more targeted.